realm-annotations-processor/src/test/java/io/realm/processor/RealmProcessorTest.java

```diff
@@ -295,7 +316,7 @@ public void compilePrimaryKeyTypes() throws IOException {
     // Unsupported "PrimaryKey" annotation types
     @Test
     public void compileInvalidPrimaryKeyTypes() throws IOException {
-        final String invalidPrimaryKeyFieldTypes[] = {"boolean", "java.util.Date", "Simple", "RealmList"};
+        final String invalidPrimaryKeyFieldTypes[] = {"boolean", "java.util.Date", "Simple", "RealmList<Simple>", "Boolean"};
```

Comment:
@cmelchior  https://github.com/realm/realm-java/issues/1486 created a issue for that.